manager.go

```diff
@@ -532,17 +532,24 @@ func (m *Manager) InitWithOptions(elf io.ReaderAt, options Options) error {
 	m.activateProbes()
 	m.state = initialized
 	m.stateLock.Unlock()
+	resetManager := func(m *Manager) {
+		m.stateLock.Lock()
+		m.state = reset
```

Comment:
This prevents any sort of cleanup if you call `Stop()`. Are we sure that there are no partial resources that need to be cleaned up if the manager errors at any of these points?

Reply:
To the best of my knowledge, before this point, and until the end of `InitWithOption()` we only do user space stuff (= parsing the elf and editing the parsed code) or kernel space stuff that will be automatically cleanup when the process exits (such as the loaded eBPF programs) or overridden if `manager.InitWithOption()` is called again (which would be necessary due to the reset state).

But yes you're right this is a dangerous game and that's also why I didn't want to make it an actual method of the manager because it would indeed prevent a cleanup.

_edit:_ I added a `m.collection.Close()` for safe measure, but I don't think it changes much (because if the manager.InitWithOption fails, we all exit afaik)

Reply:
> because if manager.InitWithOption fails, we all exit afaik

If one module fails to load, that doesn't mean the other ones will, which means we might not exit the process.

Reply:
👍🏻